common/djangoapps/student/views.py

```diff
@@ -1670,14 +1670,12 @@ def create_account_with_params(request, params):
         getattr(settings, 'REGISTRATION_EXTRA_FIELDS', {})
     )
 
-    # Boolean of whether a 3rd party auth provider and credentials were provided in
-    # the API so the newly created account can link with the 3rd party account.
-    #
-    # Note: this is orthogonal to the 3rd party authentication pipeline that occurs
-    # when the account is created via the browser and redirect URLs.
-    should_link_with_social_auth = third_party_auth.is_enabled() and 'provider' in params
+    # set random password for accounts made via third party
+    set_random_password = third_party_auth.is_enabled() and ('provider' in params or 'is_third_party_request' in params)
+    if set_random_password:
+        if 'is_third_party_request' in params and not pipeline.running(request):
```

Comment:
We can remove this check as well as validation error after getting confirmed that `pipeline.running(request)` is only used to identify that current request for account creation is made using third party auth as new flag `'is_third_party_request' in params` is now doing that part.